packages/pkh-sui/src/verifier.ts

```diff
@@ -0,0 +1,40 @@
+import {
+    SiwSuiMessage,
+    Cacao,
+    VerifyOptions,
+    verifyTimeChecks,
+    assertSigned,
+    Verifiers,
+  } from '@didtools/cacao'
+  import { AccountId } from 'caip'
+  import nacl from 'tweetnacl';
+  import { fromString as u8aFromString } from 'uint8arrays/from-string'
+  
+  /**
+   *  Get a configured CACAO SuiVerifier map for Sui accounts
+   */
+  export function getSuiVerifier(): Verifiers {
+    return {
+      // eslint-disable-next-line @typescript-eslint/require-await
+      'sui:ed25519': async (cacao: Cacao, opts: VerifyOptions): Promise<void> => {
+        verifySuiSignature(cacao, opts)
+      },
+    }
+  }
+  
+  export function verifySuiSignature(cacao: Cacao, options: VerifyOptions) {
```

Comment:
@zachferland Ive been looking into this more and it appears there is not currently a method to get the public key from the sui address to verify the signature. Im looking for insight from you to see what you would consider the best method for approaching this problem?

Reply:
is secp256k1 not also supported? then ecrecover from signature can be used (like eth)

Reply:
keys can be either ed25519, secp256k1, secp256r1

Reply:
do wallets support all three? is one default? how does that work typically

if these messages could just use secp256k1, that would be great

Reply:
Wallets can be one of the 3 and it is specified at account creation. It is set by a flag when the a keypair is created programmatically.